go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -297,10 +303,17 @@ func startOvnKube(ctx *cli.Context, cancel context.CancelFunc) error {
 	}
 
 	// no need for leader election in node mode
+	// only node mode
 	if !runMode.clusterManager && !runMode.ovnkubeController {
 		return runOvnKube(ctx.Context, runMode, ovnClientset, eventRecorder)
 	}
 
+	// ovnkube-controller with node
+	if runMode.node && runMode.ovnkubeController {
+		metrics.RegisterOVNKubeControllerBase()
+		return runOvnKube(ctx.Context, runMode, ovnClientset, eventRecorder)
+	}
```

Comment:
this change only removes leader election for ovnkube-controller+ovnkube-node config. If we want to have leader election inside one zone for ovnkube-controller AND run ovnkube-node in the same container, it will require some bigger changes anyway, and if not, we can just run separate containers as before. Or maybe you mean something else?

Reply:
I see what you mean. I thought it would be straight forward as microshift already does it but that I guess builds up on the basis that it is guaranteed to win leader election, just as we would in single node zones. So we could keep leader election on that same basis, and if we ever want HA the improvement would be make those bigger changes? Rather than remove leader election alltogether?

Reply:
yeah I can put the whole process under leader election knowing it will always win, but just so that I understand, what is the advantage of this approach?

Reply:
No much of an advantage. Let's go with what you have here.